### ISSUE TITLE
Additional parameters to tune validation of HTTPS certificate

### ISSUE THREAD

**Issue by mabrarov** [powerman/dockerize#58]
Hi.

It looks like this GitHub project is progressing faster than jwilder/dockerize, so I opened a duplicate of https://github.com/jwilder/dockerize/issues/158 here.

It looks like `no-verifyhost` option I suggested is already implemented in this project as [`skip-tls-verify` option](https://github.com/powerman/dockerize#skip-ssl-cert-verification-for-https-connections). At the same time I cannot find if `ca-file` and `verifyhost` options I suggested are implemented here, so I decided that it is still helpful suggestion of new feature.

Thank you for developing dockerize!

**Comment by powerman** [comment]
No, custom CA bundle file is not implemented. Why do you need it?

I mean, to detect when your service is up it's enough to just connect by HTTPS, make any HTTP request and get any answer. Checking is your service uses right certificate doesn't sounds like a task for dockerize. If it doesn't use right certificate then unlikely it makes sense to `-wait` anymore until wrong certificate became right certificate, huh?

**Comment by powerman** [comment]
Also, you always can install your ca.crt system-wide (e.g. into /usr/local/share/ca-certificates/) before starting dockerize, which may be a workaround in case you actually need this.

**Comment by mabrarov** [comment]
@powerman,

Thank you for quick reply.

> Checking is your service uses right certificate doesn't sounds like a task for dockerize

I'm not trying to validate certificate. My service exposes a single HTTPS endpoint so I have to use HTTPS and not HTTP. Yes, I can use `skip-tls-verify` option (and I [do that](https://github.com/mabrarov/docker-compose-init-container/pull/5)), but when working with TLS I prefer to avoid such options (some teams may be crazy about security).

> you always can install your ca.crt system-wide

This can be difficult in some cases, like non-root user and lack of ability to mount a single file into the [system-wide path](https://stackoverflow.com/questions/40051213/where-is-golang-picking-up-root-cas-from). Most of tools / runitmes I know provide an ability to specify custom CA bundle. Examples: Apache HTTP Server, Nginx, HAProxy, curl, Java. Why can't we implement the same in dockerize?

**Comment by powerman** [comment]
I've nothing against implementing it, just trying to figure out is there real use case which needs it.

So, if I get you right, the use case is "avoid skip-tls-verify everywhere by default, to even avoid thinking is it actually safe in this specific place"?

As for root - yeah, but it's **docker**ize, it supposed to run in docker, where you usually have root. But, yeah, there are some cases when you have to use some custom baseimage which already includes `USER non-root` directive in `Dockerfile`.

**Comment by mabrarov** [comment]
> "avoid skip-tls-verify everywhere by default, to even avoid thinking is it actually safe in this specific place"

Yes, sometimes it's easier and quicker solution than to prove security team that it's safe to use options like `skip-tls-verify` in specific places. I understand that it's not the best way of doing the business.

> it supposed to run in docker

I guess all of us want images and containers, but in some places the only approved platform to run containers in production environment is K8s (security team insists to use non-root user) or OpenShift (uses random UID by default, security team requires non-root user). Not Docker (even as container runtime). At the same time Docker (Docker Compose) is the most lightweight option for local and quick to setup from scratch environment. I doubt that smbd will use different images for these cases and I doubt smbd wants to develop (and so test) locally with root user but run in production with non-root (random) UID.

**Comment by powerman** [comment]
Released in 0.13.0.
